hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/StakeChangeManager.java

```diff
@@ -0,0 +1,186 @@
+package com.hedera.services.ledger.accounts.staking;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.ledger.EntityChangeSet;
+import com.hedera.services.ledger.properties.AccountProperty;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hedera.services.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.swirlds.merkle.map.MerkleMap;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static com.hedera.services.context.properties.StaticPropertiesHolder.STATIC_PROPERTIES;
+import static com.hedera.services.ledger.properties.AccountProperty.BALANCE;
+import static com.hedera.services.ledger.properties.AccountProperty.DECLINE_REWARD;
+import static com.hedera.services.ledger.properties.AccountProperty.STAKED_ID;
+import static com.hedera.services.ledger.properties.AccountProperty.STAKED_TO_ME;
+
+@Singleton
+public class StakeChangeManager {
+	private final StakeInfoManager stakeInfoManager;
+	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts;
+
+	@Inject
+	public StakeChangeManager(final StakeInfoManager stakeInfoManager,
+			final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts) {
+		this.stakeInfoManager = stakeInfoManager;
+		this.accounts = accounts;
+	}
+
+	public void withdrawStake(final long curNodeId, final long amount, final boolean declinedReward) {
+		final var node = stakeInfoManager.mutableStakeInfoFor(curNodeId);
+		if (declinedReward) {
+			node.setStakeToNotReward(node.getStakeToNotReward() - amount);
```

Comment:
_NIT:_ would prefer `node.removeRewardStake(amount)` and `node.addRewardStake(amount)`.